src/components/NavigationBar/index.jsx

```diff
@@ -41,115 +47,242 @@ function NavigationBar({ theme }) {
+                    border: 'none'
+                  },
+                  color: 'common.white'
+                }}
                 value={i18n.language}
               >
-                <option value={'en-GB'}>🇬🇧 -English</option>
-                <option value={'pt-PT'}>🇵🇹 - Português</option>
-              </select>
-            </NavDropdown>
-          </Nav>
-          <IconButton
-            color={'inherit'}
-            onClick={toggleDarkMode}
-            size={'small'}
-            sx={{
-              '&:hover': {
-                backgroundColor: '#000000'
-              },
-              backgroundColor: '#495057',
-              ml: 1
-            }}
-            variant={'contained'}
+                <MenuItem value={'pt-PT'}>🇵🇹 PT</MenuItem>
+                <MenuItem value={'en-GB'}>🇬🇧 EN</MenuItem>
+              </Select>
+
+              <IconButton
+                color={'inherit'}
+                onClick={toggleDarkMode}
+                size={'small'}
+                sx={{ '&:hover': { backgroundColor: '#000000' }, backgroundColor: '#495057' }}
+              >
+                {darkMode ? (
+                  <Brightness7
+                    fontSize={'inherit'}
+                    sx={{ color: 'common.white' }}
+                  />
+                ) : (
+                  <Brightness4
+                    fontSize={'inherit'}
+                    sx={{ color: 'common.white' }}
+                  />
+                )}
+              </IconButton>
+            </Hidden>
+
+            <Hidden mdUp>
+              <IconButton
+                aria-label={'menu'}
+                color={'inherit'}
+                edge={'end'}
+                onClick={handleDrawerOpen}
+              >
+                <MenuIcon />
+              </IconButton>
+            </Hidden>
+          </Toolbar>
+        </Container>
+      </AppBar>
+      <Drawer
+        anchor={'right'}
+        onClose={handleDrawerClose}
+        open={drawerOpen}
+      >
+        <List sx={{ width: 250 }}>
+          <ListItem
+            button
+            component={Link}
+            to={'/'}
+          >
+            <ListItemText primary={t('menu.home')} />
+          </ListItem>
+          <ListItem
+            button
+            component={Link}
+            to={'/product/list'}
           >
-            {darkMode ? (
-              <Brightness7
-                fontSize={'inherit'}
-                style={{ color: 'white' }}
-                // eslint-disable-next-line react/jsx-closing-bracket-location
-              />
-            ) : (
-              <Brightness4
-                fontSize={'inherit'}
-                style={{ color: 'white' }}
-                // eslint-disable-next-line react/jsx-closing-bracket-location
-              />
-            )}
-          </IconButton>
-        </Navbar.Collapse>
-      </Container>
-    </Navbar>
+            <ListItemText
```

Comment:
on mobile, product list LI has both elements stacked. The container for the text value and number of products needs to have display flex on it so it can space out both elements inline